plugins/reposync.py

```diff
@@ -109,10 +112,40 @@ def run(self):
             if self.opts.remote_time:
                 repo._repo.setPreserveRemoteTime(True)
             if self.opts.download_metadata:
-                self.download_metadata(repo)
+                if self.opts.urls:
+                    for md_type, md_location in repo._repo.getMetadataLocations():
+                        url = repo.remote_location(md_location)
+                        if url:
+                            print(url)
+                        else:
+                            msg = _("Failed to get mirror for metadata: %s") % md_type
+                            logger.warning(msg)
+                else:
+                    self.download_metadata(repo)
             if self.opts.downloadcomps:
-                self.getcomps(repo)
-            self.download_packages(repo)
+                if self.opts.urls:
+                    mdl = dict(repo._repo.getMetadataLocations())
+                    group_locations = [mdl[md_type]
+                                       for md_type in ('group', 'group_gz', 'group_gz_zck')
+                                       if md_type in mdl]
+                    if group_locations:
+                        for group_location in group_locations:
+                            url = repo.remote_location(group_location)
+                            if url:
+                                print(url)
+                                break
+                        else:
+                            msg = _("Failed to get mirror for the group file.")
+                            logger.warning(msg)
```

Comment:
We discussed this offline, but I want to mention it here as well to keep better track of it - this code actually works with different comps locations than the getcomps method.

Reply:
Yes, the getcomps method uses cached metadata - I can add demand `freshest_metadata` to make sure that metadata in cache are really up-to-date regardless the expiration status of the repository. It is not useful to download the whole metadata set again and we do not have method to download only comps metadata - we have only possibility to download default metadata set (primary, comps and filelists), full metadata (used by reposync --download-metadata) or  we can add some extra metadata to the default set (for example "other" metadata containing the changelogs).

But when printing the URL of the comps metadata, the cached version is not useful -  we need to find out original url from where the file was downloaded to the cache.